service/lib/ops/policyOps.js

```diff
@@ -180,16 +180,26 @@ const policyOps = {
   createPolicy: function createPolicy (params, cb) {
     const { id, version, name, organizationId, statements } = params
 
-    insertPolicies(db, [{
-      id: id,
-      version: version,
-      name: name,
-      org_id: organizationId,
-      statements: statements
-    }], (err, result) => {
-      if (err) return cb(err)
-
-      policyOps.readPolicy({ id: result.rows[0].id, organizationId }, cb)
+    const checkIdSql = SQL`
+      SELECT id
+      FROM policies
+      WHERE id = ${id}
+    `
+    db.query(checkIdSql, (err, res) => {
```

Comment:
I find this approach error prone, since another policy could be inserted between the check and the actual insertion.
I suggest two other possible approaches:
1. check the error code returned from the db: if it's a constraint key violation on id return proper error.
2. switch the insert from INSERT ... VALUES to an INSERT FROM with a where clause that check the existence of the id. In this case we can check and if any value has been inserted: if not the id is duplicate

I prefer version 1 if is possible, because is more clear and describe the intent more clearly, the second one otherwise.

Bonus point: both versions consist of a single query which should benefit performances